app/src/main/java/com/example/nextgen/splashScreen/SplashScreenActivity.kt

```diff
@@ -5,24 +5,35 @@ import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
 import androidx.lifecycle.lifecycleScope
 import com.example.nextgen.R
+import com.example.nextgen.home.HomeActivity
 import com.example.nextgen.signup.SignupSigninActivity
+import com.google.firebase.auth.FirebaseAuth
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 
 class SplashScreenActivity : AppCompatActivity() {
 
+    private lateinit var firebaseAuth: FirebaseAuth
```

Comment:
You should inject firebaseAuth instead of creating directly instance In activity, and extend splash activity from BaseActivity. Please have a look at SignUpSignInActivity for reference.